sqlglot/dialects/snowflake.py

```diff
@@ -1376,3 +1376,25 @@ def select_sql(self, expression: exp.Select) -> str:
             if offset and not limit:
                 expression.limit(exp.Null(), copy=False)
             return super().select_sql(expression)
+
+        def createable_sql(self, expression: exp.Create, locations: t.DefaultDict) -> str:
+            kind = self.sql(expression, "kind").upper()
+            is_materialized = expression.find(exp.MaterializedProperty)
+            copy_grants_property = expression.find(exp.CopyGrantsProperty)
+
+            if kind == "VIEW" and is_materialized and copy_grants_property:
+                # For materialized views, COPY GRANTS is located *before* the columns list
+                # This is in contrast to normal views where COPY GRANTS is located *after* the columns list
+                # We default CopyGrantsProperty to POST_SCHEMA which means we need to output it POST_NAME if a materialized view is detected
+                # ref: https://docs.snowflake.com/en/sql-reference/sql/create-materialized-view#syntax
+                # ref: https://docs.snowflake.com/en/sql-reference/sql/create-view#syntax
+                post_schema_properties = locations.get(exp.Properties.Location.POST_SCHEMA, [])
+                post_schema_properties.pop(post_schema_properties.index(copy_grants_property))
+
+                this_name = self.sql(expression.this, "this")
+                copy_grants = self.sql(copy_grants_property)
+                this_schema = self.schema_columns_sql(expression.this)
+
+                return f"{this_name}{self.sep()}{copy_grants}{self.sep()}{this_schema}"
```

Comment:
Should we mimic how ClickHouse generates `this_schema` here?

https://github.com/tobymao/sqlglot/blob/10fe4c039a15f12c97bdf74e2e4cf547691f8546/sqlglot/dialects/clickhouse.py#L1300-L1303

I wonder if this generates an extraneous whitespace character in case there's no schema. If this isn't even valid syntax in Snowflake though then feel free to ignore.